analytics/management/commands/check_analytics_state.py

```diff
@@ -0,0 +1,87 @@
+from argparse import ArgumentParser
+from datetime import timedelta
+
+from django.core.management.base import BaseCommand
+from django.utils.timezone import now as timezone_now
+from django.utils.timezone import utc as timezone_utc
+
+from analytics.models import InstallationCount, installation_epoch, \
+    last_successful_fill
+from analytics.lib.counts import COUNT_STATS, CountStat
+from zerver.lib.timestamp import floor_to_hour, floor_to_day, verify_UTC, \
+    TimezoneNotUTCException
+from zerver.models import Realm
+
+import os
+import subprocess
+import sys
+from typing import Any, Dict
+
+states = {
+    0: "OK",
+    1: "WARNING",
+    2: "CRITICAL",
+    3: "UNKNOWN"
+}
+
+class Command(BaseCommand):
+    help = """Checks FillState table.
+
+    Run as a cron job that runs every hour."""
+
+    def handle(self, *args, **options):
+        # type: (*Any, **Any) -> None
+        fill_state = self.get_fill_state()
+        status = fill_state['status']
+        message = fill_state['message']
+
+        state_file_path = "XXX"
+        state_file_tmp = state_file_path + "-tmp"
+
+        with open(state_file_tmp, "w") as f:
+            f.write("%s|%s|%s|%s\n" % (
+                timezone_now(), status, states[status], message))
+        subprocess.check_call(["mv", state_file_tmp, state_file_path])
+
+    def get_fill_state(self):
+        # type: () -> Dict[str, Any]
+        if not Realm.objects.exists():
+            return {'status': 0, 'message': 'No realms exist, so not checking FillState.'}
+
+        warning_unfilled_properties = []
+        critical_unfilled_properties = []
+        for property, stat in COUNT_STATS.items():
+            last_fill = last_successful_fill(property)
+            if last_fill is None:
+                last_fill = installation_epoch()
+            try:
+                verify_UTC(last_fill)
+            except TimezoneNotUTCException:
+                return {'status': 2, 'message': 'FillState not in UTC for %s' % (property,)}
+
+            if stat.frequency == CountStat.DAY:
+                floor_function = floor_to_day
+                warning_threshold = timedelta(hours=50)
+                critical_threshold = timedelta(hours=26)
+            else:  # CountStat.HOUR
+                floor_function = floor_to_hour
+                warning_threshold = timedelta(minutes=90)
+                critical_threshold = timedelta(hours=150)
```

Comment:
I think you want these to be basically "warning = missed at most 1 run", "critical = missed at least 2 runs".  It's user-facing if these things are behind.